build/bazel/remote/asset/v1/remote_asset.proto

```diff
@@ -234,6 +239,15 @@ message FetchBlobResponse {
   // The result of the fetch, if the status had code `OK`.
   // The digest of the file's contents, available for download through the CAS.
   build.bazel.remote.execution.v2.Digest blob_digest = 5;
+
+  // This field SHOULD be set to the digest function that was used by the server
+  // to compute [FetchBlobResponse.blob_digest].
+  // Clients could use this to determine whether the server honors
+  // [FetchBlobRequest.digest_function] that was set in the request.
```

Comment:
Couldn't it be a simple boolean if it's only used by the client to validate that it was respected by the server?

Reply:
I suppose it could, but it seems weird to have a field that explicitly says, "I honoured this part of your request," vs. one that says "Here's how to interpret this other part of the response (which has the incidental effect of confirming that the system honoured the digest_function of the request." Honestly it seems like an oversight that the current API doesn't qualify blob_digest with a digest_function, given the way this has evolved in the RE API.

Reply:
Yes, it's a bit unfortunate.

I will go ahead with the boolean suggestion. I too think it's ugly, but alternative routes would require more disruptive changes.

Reply:
I still think that including digest_function is the correct path here. There are already many places in the API where we pair a Digest with a digest_function so that the receiver knows how to interpret the field, and a digest_function field is useful even if the client *doesn't* specify a digest_function in the request (e.g., the server can encode SHA256 in the response to remove any ambiguity).

Ed, WDYT?

Reply:
I'm fine either way. 😀

Reply:
Reverted from `bool -> DigestFunction.Value`.

This PR is now ready to be merged.

@bergsieker you could use the Github UI to squash the commits into 1 single commit. Or I could manually squash them and push them as well.